sdk/servicebus/azure_messaging_servicebus/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All Rights reserved
+// Licensed under the MIT license.
+
+#![recursion_limit = "128"]
+#![warn(missing_docs)]
+#![doc = include_str!("../README.md")]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+
+/// Service Bus client
+pub mod client;
+mod error;
+mod message;
+/// Service Bus message receiving functionality and options.
+pub mod receiver;
+
+/// Service Bus message sending functionality and options.
+pub mod sender;
+
+/// Models and types used throughout the Service Bus client.
+pub mod models;
+
+/// Common types and utilities.
+mod common;
+
+pub use client::{
+    CreateReceiverOptions, CreateSenderOptions, ServiceBusClient, ServiceBusClientBuilder,
+    ServiceBusClientOptions, SubQueue,
+};
+pub use error::{ErrorKind, ServiceBusError};
+pub use message::{Message, MessageBatch, ReceivedMessage};
+pub use receiver::{
+    AbandonMessageOptions, CompleteMessageOptions, DeadLetterMessageOptions, DeferMessageOptions,
+    PeekMessagesOptions, ReceiveDeferredMessagesOptions, ReceiveMessageOptions, ReceiveMode,
+    Receiver, RenewMessageLockOptions,
+};
+pub use sender::{
+    CancelScheduledMessagesOptions, CreateMessageBatchOptions, ScheduleMessageOptions,
+    ScheduleMessagesOptions, SendMessageBatchOptions, SendMessageOptions, SendMessagesOptions,
+    Sender,
+};
+
+/// Result type used throughout the Service Bus client.
+pub type Result<T> = std::result::Result<T, ServiceBusError>;
```

Comment:
This should be defined in `errors.rs` but re-exported here. We want to keep all error-related stuff together.

Reply:
@LarryOsterman I think we also talked about this but never codified it in guidelines, but maybe we should allow crates to return their own error IIF it's convertible to/from `azure_core::Error`.

/cc @analogrelay who asked about this for Cosmos as well.

Reply:
As it is now, this violates the guidelines but I think we could add something like,

* You MAY add your own error and associated result if there are any actionable error kinds unique to your crate.
* You MUST return this error or associated result from all client methods if you define one.
* You MUST implement `From` to convert both to and from `azure_core::Error` if you define one.

The latter point means that people can still mix different clients and use the `?` operator for ease if their method only returns an `azure_core::Result`.